### PR TITLE
test: add SetDividendWallet test coverage

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/procedures/__tests__/SetDividendsWallet.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      // Real call
+      expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should add transactions to the queue for change wallet with a new Erc20 dividend module', async () => {
```
Comment (victorwiebe) [inline src/procedures/__tests__/SetDividendsWallet.ts]: Fede, your code is good, nice work. The only comment I have is that I feel like you should write this test two times. I think you should test both 

"should add transactions to the queue for change wallet with a new **Erc20** dividend module"
"should add transactions to the queue for change wallet with a new **ETH** dividend module"

File: src/procedures/SetDividendsWallet.ts [hunk 2]
```diff
@@ -47,6 +58,9 @@ export class SetDividendsWallet extends Procedure<SetDividendsWalletProcedureArg
         );
         break;
       }
+      default: {
```
Comment (monitz87) [inline src/procedures/SetDividendsWallet.ts]: Why was this added?

File: src/procedures/SetDividendsWallet.ts [hunk 3]
```diff
@@ -76,6 +90,9 @@ export class SetDividendsWallet extends Procedure<SetDividendsWalletProcedureArg
               })
             );
           }
+          default: {
```
Comment (monitz87) [inline src/procedures/SetDividendsWallet.ts]: Why was this added?

File: src/procedures/__tests__/SetDividendsWallet.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe(ProcedureType.SetDividendsWallet);
+    });
+  });
+
+  describe('SetDividendsWallet', () => {
+    test('should throw if there is no valid dividend type being provided', async () => {
+      // Instantiate SetDividendsWallet with incorrect dividend type
+      target = new SetDividendsWallet(
+        {
```
Comment (monitz87) [inline src/procedures/__tests__/SetDividendsWallet.ts]: You can do

```
{
  ...params,
  dividendType: 'wrong' as DividendType,
}
```

File: src/procedures/__tests__/SetDividendsWallet.ts (hunk 1 above)
Comment (monitz87) [reply src/procedures/__tests__/SetDividendsWallet.ts]: Agreed

Comment (monitz87) [reply src/procedures/__tests__/SetDividendsWallet.ts]: The description text is a bit off though. Should be

"should add a transaction to the queue to change the dividends wallet of an attached ERC20 dividends module"

And the same for ETH

File: src/procedures/SetDividendsWallet.ts (hunk 2 above)
Comment (shuffledex) [reply src/procedures/SetDividendsWallet.ts]: TypeScript recommendation

File: src/procedures/SetDividendsWallet.ts (hunk 3 above)
Comment (shuffledex) [reply src/procedures/SetDividendsWallet.ts]: TypeScript recommendation

File: src/procedures/SetDividendsWallet.ts (hunk 2 above)
Comment (monitz87) [reply src/procedures/SetDividendsWallet.ts]: What do you mean by that? Which tool recommended it exactly? The compiler says nothing

File: src/procedures/__tests__/SetDividendsWallet.ts [hunk 5]
```diff
@@ -69,6 +97,28 @@ describe('SetDividendsWallet', () => {
...
 
   describe('SetDividendsWallet', () => {
+    test('should throw if there is no valid security token being provided', async () => {
+      // Instantiate SetDividendsWallet with incorrect security symbol
+      target = new SetDividendsWallet(
+        { ...params, dividendType: DividendType.Erc20 },
+        contextMock.getMockInstance()
+      );
+
+      tokenFactoryMock.set(
```
Comment (monitz87) [inline src/procedures/__tests__/SetDividendsWallet.ts]: This mock is badly written (see slack for corrections)

File: src/procedures/__tests__/SetDividendsWallet.ts [hunk 6]
```diff
@@ -69,6 +97,28 @@ describe('SetDividendsWallet', () => {
...
+        'getSecurityTokenInstanceFromTicker',
+        stub()
+          .withArgs({ address: params.symbol })
+          .throws()
+      );
+
+      expect(target.prepareTransactions()).rejects.toThrow(
```
Comment (monitz87) [inline src/procedures/__tests__/SetDividendsWallet.ts]: This should be awaited

File: src/procedures/__tests__/SetDividendsWallet.ts [hunk 7]
```diff
@@ -177,5 +227,51 @@ describe('SetDividendsWallet', () => {
       ).toEqual(true);
       expect(addTransactionSpy.callCount).toEqual(1);
     });
+
+    test('should successfully refresh ERC20 dividends factory', async () => {
+      const refreshStub = erc20FactoryMock.mock(
+        'refresh',
+        Promise.resolve(undefined)
+      );
+      const resolverValue = setDividendsWalletModule.createSetDividendWalletResolver(
```
Comment (monitz87) [inline src/procedures/__tests__/SetDividendsWallet.ts]: This should be awaited here to make sure there is no chance for the stubs to not have been called by the time the verifications are made

File: src/procedures/__tests__/SetDividendsWallet.ts [hunk 8]
```diff
@@ -177,5 +227,51 @@ describe('SetDividendsWallet', () => {
...
+      expect(refreshStub.callCount).toEqual(1);
+    });
+  });
+
+  test('should successfully refresh Eth dividends factory', async () => {
+    const refreshStub = ethFactoryMock.mock(
+      'refresh',
+      Promise.resolve(undefined)
+    );
+    const resolverValue = setDividendsWalletModule.createSetDividendWalletResolver(
```
Comment (monitz87) [inline src/procedures/__tests__/SetDividendsWallet.ts]: This should be awaited here to make sure there is no chance for the stubs to not have been called by the time the verifications are made
